src/lib/browse/PointAmenityLayerControl.svelte

```diff
@@ -0,0 +1,61 @@
+<script lang="ts">
+  import type { FilterSpecification, MapGeoJSONFeature } from "maplibre-gl";
+  import {
+    isPoint,
+    overwriteCircleLayer,
+    overwriteGeojsonSource,
+  } from "../../maplibre_helpers";
+  import { map } from "../../stores";
+  import { ColorLegend, HelpButton, InteractiveLayer } from "../common";
+  import { Checkbox } from "../govuk";
+  import { colors } from "./colors";
+
+  // This name is used for multiple things:
+  // - The name of a .pmtiles file
+  // - The layer name within that file
+  // - The layer name, for layerZorder
+  // - A color name in colors.ts
+  export let name: string;
+  // Uncapitalized
+  export let singularNoun: string;
+  // Capitalized
+  export let pluralNoun: string;
+
+  export let circleRadius: number;
+
+  // The caller must also fill in the default slot with the contents of a help modal
+
+  // @ts-ignore TODO Also constrain name to exist in the colors type
+  let color = colors[name];
+
+  const layerLoadedCallback = () => {
+    overwriteCircleLayer($map, {
+      id: name,
+      source: name,
+      filter: ["all", isPoint] as FilterSpecification,
```

Comment:
For our one use case so far of railway stations, this should be unnecessary. We should make sure the input data only contains points in the first place, or understand what stations aren't available as single points and deal with them somehow. Otherwise this is just hiding a problem of missing data

Reply:
I wrongly thought you needed a filter and it wouldn't let me have an empty list. I really think if you have something that is represented as a list of some kind that you should represent its absence as an empty list. But that's by the by, removing

Reply:
That's an unfortunate API choice in MapLibre styles generally. In methods where we take optional parameters, we use [defaults](https://github.com/acteng/atip/blob/3c844aeaaeb2acd5a1f9f91e050fe3bef4edb32c/src/maplibre_helpers.ts#L18) like this, because that's seemingly the only choice